addons/mail/static/src/models/media_preview/media_preview.js

```diff
@@ -0,0 +1,173 @@
+         * method on each of them.
+         * 
+         * @static
+         * @param {MediaStream} mediaStream 
+         */
+        static stopTracksOnMediaStream(mediaStream) {
+            for (const track of mediaStream.getTracks()) {
+                track.stop();
+            }
+        }
+
+        /**
+         * Stops recording user's microphone.
+         */
+        disableMicrophone() {
+            this.audioRef.el.srcObject = null;
+            if (!this.audioStream) {
+                return;
+            }
+            this.messaging.models['mail.media_preview'].stopTracksOnMediaStream(this.audioStream);
+            this.update({ audioStream: null });
+        }
+
+        /**
+         * Stops recording user's video device.
+         */
+        disableVideo() {
+            this.videoRef.el.srcObject = null;
+            if (!this.videoStream) {
+                return;
+            }
+            this.messaging.models['mail.media_preview'].stopTracksOnMediaStream(this.videoStream);
+            this.update({ videoStream: null });
+        }
+
+        /**
+         * Asks for access to the user's microphone if not granted yet, then
+         * starts recording and defines the resulting audio stream as the source
+         * of the audio element in order to play the audio feedback.
+         */
+        async enableMicrophone() {
+            if (!this.doesBrowserSupportMediaDevices) {
+                return;
+            }
+            const audioStream = await navigator.mediaDevices.getUserMedia({ audio: true });
+            this.update({ audioStream });
+            this.audioRef.el.srcObject = this.audioStream;
+        }
+
+        /**
+         * Asks for access to the user's video device if not granted yet, then
+         * starts recording and defines the resulting video stream as the source
+         * of the video element in order to display the video feedback.
+         */
+        async enableVideo() {
+            if (!this.doesBrowserSupportMediaDevices) {
+                return;
+            }
+            const videoStream = await navigator.mediaDevices.getUserMedia({ video: true });
+            this.update({ videoStream });
+            this.videoRef.el.srcObject = this.videoStream;
+        }
+
+        /**
+         * Handles click on the "disable microphone" button.
+         */
+        onClickDisableMicrophoneButton() {
+            this.disableMicrophone();
+        }
+
+        /**
+         * Handles click on the "disable video" button.
+         */
+        onClickDisableVideoButton() {
+            this.disableVideo();
+        }
+
+        /**
+         * Handles click on the "enable microphone" button.
+         */
+        onClickEnableMicrophoneButton() {
+            this.enableMicrophone();
+        }
+
+        /**
+         * Handles click on the "enable video" button.
+         */
+        onClickEnableVideoButton() {
+            this.enableVideo();
+        }
```

Comment:
I'm not sure if the indirection is actually useful in this case, I feel like just having the code in the handlers directly would be more readable.